cmd/discovery/discovery_integration_test.go

```diff
@@ -0,0 +1,192 @@
+
+		ns := createTestNamespace(t, k8sClient)
+		pod := createRunningPod(t, k8sClient, ns.Name)
+
+		discoverer := discovery.NewDiscoverer([]string{pod.Namespace}, kubelet)
+		discovered, err := discoverer.Run()
+		require.NoErrorf(t, err, "not expecting error when running discovery with valid namespaces")
+
+		var resultToTest discovery.DiscoveredItem
+		for _, d := range discovered {
+			if d.MetricAnnotations["podName"] == pod.Name {
+				resultToTest = d
+			}
+		}
+		require.Len(t, discovered, 1, "we are expecting only one pod")
+		require.NotEqualf(t, resultToTest, discovery.DiscoveredItem{}, "we expect resultToTest to be populated")
+		testPodIntegration(t, resultToTest, pod)
+	})
+
+	t.Run("succeeds_with_not_existing_namespace", func(t *testing.T) {
+		t.Parallel()
+
+		ns := createTestNamespace(t, k8sClient)
+		createRunningPod(t, k8sClient, ns.Name)
+
+		discoverer := discovery.NewDiscoverer([]string{"not-existing"}, kubelet)
+		discovered, err := discoverer.Run()
+		require.NoErrorf(t, err, "not expecting error when running discovery with not-existing namespaces")
+
+		require.Len(t, discovered, 0, "we are not expecting any element")
+	})
+
+	t.Run("succeeds_with_failing_pod", func(t *testing.T) {
+		t.Parallel()
+
+		ns := createTestNamespace(t, k8sClient)
+
+		pod := getPodIntegration()
+		pod.Spec.Containers[0].Image = "not-existing-image"
+		pod, err := k8sClient.CoreV1().Pods(ns.Name).Create(pod)
+		require.NoErrorf(t, err, "not expecting error when creating test pod")
+
+		discoverer := discovery.NewDiscoverer([]string{"not-existing"}, kubelet)
+		discovered, err := discoverer.Run()
+		require.NoErrorf(t, err, "not expecting error when running discovery with failing pods")
+
+		require.Len(t, discovered, 0, "we are not expecting any element, since failing pods should be ignored")
+	})
+}
+
+func createRunningPod(t *testing.T, k8sClient *k8s.Clientset, namespace string) *corev1.Pod {
+	pod, err := k8sClient.CoreV1().Pods(namespace).Create(getPodIntegration())
+	require.NoErrorf(t, err, "not expecting error when creating test pod")
+
+	err = retry.Do(func() error {
+		pod, err = k8sClient.CoreV1().Pods(namespace).Get(pod.Name, metav1.GetOptions{})
+		require.NoErrorf(t, err, "not expecting error getting pod")
+		if pod.Status.Phase != corev1.PodRunning {
+			return fmt.Errorf("the pod is still not running")
+		}
+		return nil
+	},
+	)
+	require.NoErrorf(t, err, "not expecting error when waiting for the pod to be running")
+
+	return pod
+}
+
+func createTestNamespace(t *testing.T, k8sClient *k8s.Clientset) *corev1.Namespace {
+	ns, err := k8sClient.CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-",
+		},
+	})
+	require.NoErrorf(t, err, "not expecting error when creating test namespace")
+
+	t.Cleanup(func() {
+		err := k8sClient.CoreV1().Namespaces().Delete(ns.Name, &metav1.DeleteOptions{})
+		require.NoErrorf(t, err, "not expecting error when cleaning environment")
+	},
+	)
```

Comment:
```suggestion
	})
```

I think this is more idiomatic formatting.